### PR TITLE
Activ 486

### PR DESCRIPTION
**Pull request by ErnestoLoma**
Rename all references to Classification and classification to Tag and tag (TagDef, Tag, tags, tagNames, etc):

### CODE REVIEW

File: pyatlan/generator/templates/methods/attribute/purpose.jinja2 [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
+        def create(cls, name: str, atlan_tags: list[str]) -> {{ entity_def.name }}.Attributes:
+            validate_required_fields(["name", "atlan_tags"], [name, atlan_tags])
             return Purpose.Attributes(
                 qualified_name=name,
                 name=name,
                 display_name=name,
                 is_access_control_enabled=True,
                 description="",
-                purpose_classifications=classifications
+                purpose_classifications=atlan_tags
```
Comment (cmgrote) [inline pyatlan/generator/templates/methods/attribute/purpose.jinja2:12]: Can we alias `purpose_classifications` here, too?

File: pyatlan/generator/templates/referenceable_attributes.jinja2 [hunk 2]
```diff
@@ -72,8 +72,8 @@
     version: Optional[int] = Field(
         None, description='Version of this object.\n', example=2
     )
-    classifications: Optional[list[Classification]] = Field(
-        None, description="classifications"
+    atlan_tags: Optional[list[AtlanTag]] = Field(
+        None, description="classifications", alias="classifications"
```
Comment (cmgrote) [inline pyatlan/generator/templates/referenceable_attributes.jinja2:76]: I would change the description to "Atlan tags" as well — we should only have `classification(s)` appear as an alias, and literally nowhere else (if possible)

File: tests/integration/purpose_test.py [hunk 3]
```diff
@@ -24,22 +24,21 @@
 @pytest.fixture(scope="module")
 def atlan_tag(
     client: AtlanClient,
-) -> Generator[ClassificationDef, None, None]:
-    classification_def = ClassificationDef.create(
+) -> Generator[AtlanTagDef, None, None]:
+    classification_def = AtlanTagDef.create(
```
Comment (cmgrote) [inline tests/integration/purpose_test.py:28]: I'd:
1. Change the variable name to `atlan_tag_def` for consistency
2. Change the name of the enum from `AtlanClassificationColor` to `AtlanTagColor`

**Review by cmgrote — CHANGES_REQUESTED** [review]
Also, in the `Tag` typedef there is an attribute called `mappedClassificationName` which I think we need to rename to `mappedAtlanTagName` (and alias back to the original name)

File: tests/integration/purpose_test.py (hunk 3 above)
Comment (ErnestoLoma) [reply tests/integration/purpose_test.py:28]: done

File: pyatlan/generator/templates/referenceable_attributes.jinja2 (hunk 2 above)
Comment (ErnestoLoma) [reply pyatlan/generator/templates/referenceable_attributes.jinja2:76]: done

**Review by cmgrote — APPROVED** [review]

**Pull request merged**